src/pages/try.js

```diff
@@ -21,32 +21,45 @@ if (typeof navigator !== 'undefined') {
     </div>
   )
 }
-const reasonQueryParamPrefix = "?reason="
-const ocamlQueryParamPrefix = "?ocaml="
-
-// returns [code, isReason]
-const decodeSnippetFromURL = () => {
-  const queryParam = window.location.search; // returns ?encoded_snippet=blablabla
-  if (queryParam.startsWith(reasonQueryParamPrefix)) {
-    const encodedSnippet = queryParam.slice(reasonQueryParamPrefix.length);
-    return [decompress(encodedSnippet), true];
-  } else if (queryParam.startsWith(ocamlQueryParamPrefix)) {
-    const encodedSnippet = queryParam.slice(ocamlQueryParamPrefix.length);
-    return [decompress(encodedSnippet), false];
-  } else {
-    return ['let x = 10;\nJs.log x;', true]
+const  queryParamPrefixFor = language => `?${language}=`;
+
+const retrieve = () => {
+  function fromQueryParam(language) {
+    const queryParam = window.location.search; // returns ?language=blablabla
+    const prefix = queryParamPrefixFor(language);
+
+    if (queryParam.startsWith(prefix)) {
+      return {
+        language,
+        code: decompress(queryParam.slice(prefix.length)) || '' // decompressing an empty string returns null, joyously!
```

Comment:
do you know if decompress ever throws?

Reply:
It doesn't seem to, it just returns an empty string on invalid input. We could still detect that and default to something more useful though.